Document registering secured APIs through Metadata Service (#1699)
kyma-project/kyma: docs/application-connector/docs/037-gs-register-secured-api.md

---
title: Register a secured API
type: Getting Started
---

> 

Reviewer: [GS template](https://github.com/kyma-project/community/blob/master/guidelines/templates/resources/getting-started.md)?

Author: Right, this will work better as a "details" doc.

The Metadata Service allows you to register a secured API for every service. The supported authentication methods are [Basic Authentication](https://tools.ietf.org/html/rfc7617) and [OAuth](https://tools.ietf.org/html/rfc6750).

> You can specify only one authentication method for every secured API you register. If you try to register and specify more than one authentication method, the Metadata Service returns a code `400` response.

Reviewer: ```suggestion
You can specify only one authentication method for every secured API you register. If you try to register and specify more than one authentication method, the Metadata Service returns the `400` status code.
```

Author: The service doesn't return a code, it returns a response which is identified by its code.

Reviewer: a `400` code response then?

> ## Register an Basic Authentication-secured API

Reviewer: ```suggestion
## Register a Basic Authentication-secured API
```

To register an API secured with Basic Authentication, add a `credentials.basic` object to the `api` section of the service registration request body. You must include these fields:

| Field   |  Description |
|:----------:|:------|

> | **username** | Basic Authorization username. |

Reviewer: ```suggestion
| **username** | Basic Authorization username |
```

> | **password** | Basic Authorization password. |

Reviewer: ```suggestion
| **password** | Basic Authorization password |
```

This is an example of the `api` section of the request body for an API secured with Basic Authentication:

```
    "api": {
        "targetUrl": "https://sampleapi.targeturl/v1",
        "credentials": {
            "basic": {
                "username": "{USERNAME}",
                "password": "{PASSWORD}"
            },
        }  
```
## Register an OAuth-secured API

To register an API secured with OAuth, add a `credentials.oauth` object to the `api` section of the service registration request body. You must include these fields:

| Field   |  Description |
|:----------:|:------|

> | **url** |  OAuth token exchange endpoint of the service. |

Reviewer: ```suggestion
| **url** |  OAuth token exchange endpoint of the service |
```

> | **clientId** | OAuth client ID. |

Reviewer: ```suggestion
| **clientId** | OAuth client ID |
```

> | **clientSecret** | OAuth client Secret. |    

Reviewer: ```suggestion
| **clientSecret** | OAuth client Secret |    
```
